pkg/scheduler/core/generic_scheduler.go

```diff
@@ -360,13 +366,30 @@ func (g *genericScheduler) findNodesThatPassExtenders(pod *v1.Pod, feasibleNodes
 			return nil, err
 		}
 
+		for failedNodeName, failedMsg := range failedAndUnresolvableMap {
+			if _, found := statuses[failedNodeName]; !found {
+				statuses[failedNodeName] = framework.NewStatus(framework.UnschedulableAndUnresolvable, failedMsg)
+			} else {
+				if statuses[failedNodeName].Code() != framework.UnschedulableAndUnresolvable {
```

Comment:
This `if` should be always true, right? If so, how about refactor the whole logic to be:

```go
for failedNodeName, failedMsg := range failedAndUnresolvableMap {
	var aggregatedReasons []string
	if _, found := statuses[failedNodeName]; found {
		aggregatedReasons = statuses[failedNodeName].Reasons()
	}
	aggregatedReasons = append(aggregatedReasons, failedMsg)
	statuses[failedNodeName] = framework.NewStatus(framework.UnschedulableAndUnresolvable, aggregatedReasons...)
}
```

Reply:
make sense, updated